src/cpp/fastdds/builtin/type_lookup_service/TypeLookupRequestListener.cpp

```diff
@@ -50,94 +101,179 @@ TypeLookupRequestListener::~TypeLookupRequestListener()
 {
 }
 
+void TypeLookupRequestListener::check_get_types_request(
+        SampleIdentity request_id,
+        const TypeLookup_getTypes_In& request)
+{
+    TypeLookup_getTypes_Out out;
+    // Iterate through requested type_ids
+    for (const xtypes::TypeIdentifier& type_id : request.type_ids())
+    {
+        xtypes::TypeObject obj;
+        if (RETCODE_OK == fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                        get_type_object(type_id, obj))
```

Comment:
The remote exception code should be used in case that the TypeIdentifier received is not correct.

Reply:
added exception in [29681bc](https://github.com/eProsima/Fast-DDS/pull/4240/commits/29681bc3ecb5a0cf1f4e298b7bf093bff1ab1753)